src/glue/SessionHL.cpp

```diff
@@ -167,6 +172,33 @@ bool SessionHL::Authenticate(const std::string &username, const std::string &pas
   return res;
 }
 
+bool SessionHL::SSOAuthenticate(const std::string &scheme, const std::string &identity_provider_response) {
+  bool res = true;
+  interpreter_.ResetUser();
+  {
+    auto locked_auth = auth_->Lock();
+    if (locked_auth->AccessControlled() && FLAGS_auth_sso_on) {
+      const auto user_or_role = locked_auth->SSOAuthenticate(scheme, identity_provider_response);
+      if (user_or_role.has_value()) {
+        user_or_role_ = AuthChecker::GenQueryUser(auth_, *user_or_role);
+        interpreter_.SetUser(AuthChecker::GenQueryUser(auth_, *user_or_role));
+      } else {
+        res = false;
+      }
+    } else {
+      // No access control -> give empty user
+      user_or_role_ = AuthChecker::GenQueryUser(auth_, std::nullopt);
+      interpreter_.SetUser(AuthChecker::GenQueryUser(auth_, std::nullopt));
```

Comment:
Aren't we disabling userless logins when using SSO?

Reply:
Done